### PR TITLE
V6 👀

### LINKED ISSUE #45: Upgrade bouquet to ethers v6

### PR DESCRIPTION
**Pull request by 0xjimmy**
Fix #45

### CODE REVIEW

File: app/ts/components/Import.tsx [hunk 1]
```diff
@@ -8,6 +7,7 @@ import { AppSettings, Bundle, serialize, Signers } from '../types/types.js'
 import { EthereumAddress } from '../types/ethereumTypes.js'
 import { TransactionList } from '../types/bouquetTypes.js'
 import { ImportModal } from './ImportModal.js'
+import { getAddress } from 'ethers'
```
Comment (MicahZoltu) [inline app/ts/components/Import.tsx:10]: Super nitpicky style: I personally prefer sorting external imports above internal imports, like you had originally.  🤷

File: app/ts/components/Transactions.tsx [hunk 2]
```diff
@@ -72,13 +70,13 @@ export const Transactions = ({
 				uniqueAddresses.map((address) =>
 					fetch(
 						`https://api${appSettings.peek().relayEndpoint === MEV_RELAY_GOERLI ? '-goerli' : ''
-						}.etherscan.io/api?module=contract&action=getabi&address=${utils.getAddress(address.toLowerCase())}&apiKey=PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8`,
+						}.etherscan.io/api?module=contract&action=getabi&address=${getAddress(address.toLowerCase())}&apiKey=PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8`,
```
Comment (MicahZoltu) [inline app/ts/components/Transactions.tsx:73]: API Key embedded in websites rarely ends well.  I'm guessing this is unavoidable?

File: app/ts/library/provider.ts [hunk 3]
```diff
@@ -71,28 +73,38 @@ export const connectBrowserProvider = async (
...
+		}
+
+		const [accounts, block] = await Promise.all([provider.listAccounts(), provider.getBlock('latest')])
+		if (accounts.length > 0 && window.ethereum) {
+			clearEvents()
+			window.ethereum.on('disconnect', disconnectEventCallback)
+			window.ethereum.on('accountsChanged', accountsChangedCallback)
+			window.ethereum.on('chainChanged', chainChangedCallback)
+			provider.on('block', blockCallback)
 		}
```
Comment (MicahZoltu) [inline app/ts/library/provider.ts:98]: This feels unrelated to the move to Ethers v6.  If that is the case, can it be extracted into a separate PR so it can be evaluated separately?

File: package.json [hunk 4]
```diff
@@ -14,7 +14,7 @@
 	},
 	"dependencies": {
 		"@preact/signals": "1.1.1",
-		"ethers": "5.7.2",
+		"ethers": "^6.6.1",
```
Comment (MicahZoltu) [inline package.json:17]:
```suggestion
		"ethers": "6.6.1",
```

File: package-lock.json [hunk 5]
```diff
@@ -1275,137 +187,44 @@
...
-			"integrity": "sha512-s0IrSOzLlbvX7yp4WBfPITzpAU8sqQcpsmwXDiKwrG4r491vwCO/XpejasRNl0piBMe/DvP4Tz0mIS/X1DPJBQ=="
-		},
-		"bn.js": {
-			"version": "5.2.1",
-			"resolved": "https://registry.npmjs.org/bn.js/-/bn.js-5.2.1.tgz",
-			"integrity": "sha512-eXRvHzWyYPBuB4NBy0cmYQjGitUrtqwbvlzP3G6VFnNRbsZQIxQ10PbKKHt8gZ/HW/D/747aDl+QkDqg3KQLMQ=="
+		"@types/node": {
+			"version": "18.15.13",
+			"resolved": "https://registry.npmjs.org/@types/node/-/node-18.15.13.tgz",
+			"integrity": "sha512-N+0kuo9KgrUQ1Sn/ifDXsvg0TTleP7rIy4zOBGECxAljqvqfqpTfzx0Q1NUedOixRMBfe2Whhb056a42cWs26Q=="
```
Comment (MicahZoltu) [inline package-lock.json:193]: Unfortunately until Ethers fixes their transitive dependency on `@types/node`, we'll end up grabbing this as a transitive dependency.  To deal with this, we will need to add some magic to `tsconfig.json` to get it to ignore global types from `@types/node`.  This is super annoying to do if you actually want global types for some other dependency, but you can see an example of how to do this here:
https://github.com/Zoltu/preact-es2015-template/blob/master/tsconfig.json#L28-L30

**Review by MicahZoltu — DISMISSED** [review]
Would like to see the `@node/types` issue resolved before this is merged, as well as pinning Ethers version.

I would also like to see the new code moved to a separate PR unless it is in fact part of the migration to ethers v6.

File: app/ts/components/Transactions.tsx (hunk 2 above)
Comment (0xjimmy) [reply app/ts/components/Transactions.tsx:73]: Yeah, its also taken from ABI Ninja

File: app/ts/library/provider.ts (hunk 3 above)
Comment (0xjimmy) [reply app/ts/library/provider.ts:98]: Had to change some of the types and methods when updating to v6, I also grabbed this bit from the provider.ts in nft-wallet is already in ethers v6. There was a bug when network changes the other events stop working, so assuming the the user is still connected we just clear and re-create the event listeners.

File: app/ts/library/provider.ts [hunk 6]
```diff
@@ -1,38 +1,40 @@
...
 ) => {
 	const [signer, network] = await Promise.all([provider.getSigner(), provider.getNetwork()])
 	const address = await signer.getAddress()
 	if (store.peek()) removeProvider(store)
 
-	if (![1, 5].includes(network.chainId)) {
+	const parsedAddress = AddressParser.parse(getAddress(address))
+	if (!parsedAddress.success) throw new Error("Provider provided invalid address!")
+
+	if (![1n, 5n].includes(network.chainId)) {
```
Comment (KillariDev) [inline app/ts/library/provider.ts:28]: We could make separate const for these networks as these things seem to be repeated in multiple places. You could then join chain Id, chain name and provider relay in a single variable

**Review by jubalm — APPROVED** [review]
